From 544dcf1246fc79ced7506a079fd0520585dd6827 Mon Sep 17 00:00:00 2001
From: Crabbedhaloablution <crabbedhaloablution@icloud.com>
Date: Tue, 17 Nov 2020 19:55:33 +0100
Subject: [PATCH 55/55] Add defines needed for rpi V4L2 formats

Needed for patch:
Add rpi V4L2 formats and multi-thread requests
Inspiration:
https://github.com/raspberrypi/linux/pull/3505
---
 libavcodec/v4l2_request.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/libavcodec/v4l2_request.c b/libavcodec/v4l2_request.c
index 4ca42d29ec..131fea439f 100644
--- a/libavcodec/v4l2_request.c
+++ b/libavcodec/v4l2_request.c
@@ -31,6 +31,17 @@
 #include "v4l2_request.h"
 #include "v4l2_phase.h"
 
+// V4L2_PIX_FMT_NV12_COL128 and V4L2_PIX_FMT_NV12_10_COL128 should be defined in linux/videodev2.h
+// and hopefully will be sometime in the future but until then...
+#ifndef V4L2_PIX_FMT_NV12_COL128
+#define V4L2_PIX_FMT_NV12_COL128 v4l2_fourcc('N', 'C', '1', '2') /* 12  Y/CbCr 4:2:0 128 pixel wide column */
+#endif
+
+#ifndef V4L2_PIX_FMT_NV12_10_COL128
+#define V4L2_PIX_FMT_NV12_10_COL128 v4l2_fourcc('N', 'C', '3', '0')
+                                                                /* Y/CbCr 4:2:0 10bpc, 3x10 packed as 4 bytes in
+                                                                 * a 128 bytes / 96 pixel wide column */
+#endif
 // P030 should be defined in drm_fourcc.h and hopefully will be sometime
 // in the future but until then...
 #ifndef DRM_FORMAT_P030
-- 
2.29.2

